src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -813,6 +818,74 @@ void symbolValidator(ModuleSymbol module)
         }
     }
 
+    private void DeterminismCore(
+        string fileTypeName,
+        CSharpCompilation comp1,
+        CSharpCompilation comp2,
+        CSharpCompilation comp3)
+    {
+        var outer1 = comp1.GetMember<NamedTypeSymbol>(fileTypeName).AssociatedFileIdentifier;
+        var outer2 = comp2.GetMember<NamedTypeSymbol>(fileTypeName).AssociatedFileIdentifier;
+        var outer3 = comp3.GetMember<NamedTypeSymbol>(fileTypeName).AssociatedFileIdentifier;
+        Assert.False(outer1.FilePathChecksumOpt.IsDefaultOrEmpty);
+        Assert.False(outer2.FilePathChecksumOpt.IsDefaultOrEmpty);
+        Assert.False(outer3.FilePathChecksumOpt.IsDefaultOrEmpty);
+        Assert.False(outer1.FilePathChecksumOpt.SequenceEqual(outer2.FilePathChecksumOpt));
+        Assert.True(outer1.FilePathChecksumOpt.SequenceEqual(outer3.FilePathChecksumOpt));
+
+        var emitOptions = EmitOptions.Default.WithDebugInformationFormat(DebugInformationFormat.Embedded);
+        var bytes1 = comp1.EmitToArray(emitOptions);
+        var bytes2 = comp2.EmitToArray(emitOptions);
+        var bytes3 = comp3.EmitToArray(emitOptions);
+        Assert.False(bytes1.SequenceEqual(bytes2));
+        Assert.True(bytes1.SequenceEqual(bytes3));
+    }
+
+    [Fact]
+    public void Determinism_01()
+    {
+        var source = """
+            file class Outer { }
+            """;
+
+        var (root1, root2) = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? (@"q:\", @"j:\") : ("/q", "/j");
+        var testSource1 = CSharpTestSource.Parse(source, Path.Combine(root1, "code.cs"));
+        var testSource2 = CSharpTestSource.Parse(source, Path.Combine(root2, "code.cs"));
+        var options = TestOptions.DebugDll.WithDeterministic(true);
+        var comp1 = CreateCompilation(testSource1, options: options, assemblyName: "filetypes");
+        var comp2 = CreateCompilation(testSource2, options: options, assemblyName: "filetypes");
+
+        var resolver = new SourceFileResolver(
+            ImmutableArray<string>.Empty,
+            baseDirectory: null,
+            ImmutableArray.Create(new KeyValuePair<string, string>(root2, root1)));
+        var comp3 = CreateCompilation(testSource2, options: options.WithSourceReferenceResolver(resolver), assemblyName: "filetypes");
```

Comment:
Can most of this method body be moved into `DeterminismCore()`? And if so, could we collapse the methods into a single test method with `[Theory]`?
```
[InlineData("file class Outer { }", "Outer")]
[InlineData("file class Outer1 { } file class Outer2 { }", "Outer2")]
```